core/state/journal.go

```diff
@@ -153,34 +166,30 @@ type (
 
 func (ch createObjectChange) revert(s *StateDB) {
 	delete(s.stateObjects, *ch.account)
-	delete(s.stateObjectsDirty, *ch.account)
 }
 
 func (ch createObjectChange) dirtied() *common.Address {
 	return ch.account
 }
 
-func (ch resetObjectChange) revert(s *StateDB) {
-	s.setStateObject(ch.prev)
-	if !ch.prevdestruct {
-		delete(s.stateObjectsDestruct, ch.prev.address)
-	}
-	if ch.prevAccount != nil {
-		s.accounts[ch.prev.addrHash] = ch.prevAccount
-	}
-	if ch.prevStorage != nil {
-		s.storages[ch.prev.addrHash] = ch.prevStorage
-	}
-	if ch.prevAccountOriginExist {
-		s.accountsOrigin[ch.prev.address] = ch.prevAccountOrigin
-	}
-	if ch.prevStorageOrigin != nil {
-		s.storagesOrigin[ch.prev.address] = ch.prevStorageOrigin
+func (ch createObjectChange) copy() journalEntry {
+	return createObjectChange{
+		account: ch.account,
 	}
 }
 
-func (ch resetObjectChange) dirtied() *common.Address {
-	return ch.account
+func (ch createContractChange) revert(s *StateDB) {
+	s.stateObjects[ch.account].created = false
+}
+
+func (ch createContractChange) dirtied() *common.Address {
+	return &ch.account
```

Comment:
Should we mark this account as dirty only by `createContractChange` change?

IIUC, the account is dirty because of other operations instead of this one (e.g. SetCode, CreateAccount).

Reply:
Well spotted!